frontend/src/components/MapView.jsx

```diff
@@ -1,195 +1,179 @@
+        (error) => {
+          console.error("Geolocation error:", error);
         }
-    }, [currentPosition]);
-
-    const handlePlaceSelected = (place) => {
-        handleSetLocation({
-            lat: place.geometry.location.lat(),
-            lng: place.geometry.location.lng(),
-        });
-        setZoom(15);
-        setRecentSearches([...recentSearches, {
-            name: place.name,
-            lat: place.geometry.location.lat(),
-            lng: place.geometry.location.lng(),
-        }]);
+      );
     }
+  }, []);
+
+  useEffect(() => {
+    if (isLoaded && map && currentPosition && window.google.maps) {
+      const { AdvancedMarkerElement } = window.google.maps.marker;
+      if (marker) {
+        marker.setMap(null);
+      }
+      const advancedMarker = new AdvancedMarkerElement({
+        position: currentPosition,
+        map: map,
+        title: "Default Location",
+      });
+      setMarker(advancedMarker);
+    }
+  }, [currentPosition]);
 
-    const handleSetLocation = (location) => {
-        setCurrentPosition({
-            lat: location.lat,
-            lng: location.lng,
-        });
-        setZoom(15);
-        setInfoWindowPosition({
-            lat: location.lat,
-            lng: location.lng,
-        });
-        setInfoWindowVisible(true);
-    };
-
-    const handleMapClick = () => {
-        if (marker) {
-            marker.setMap(null);
-            setMarker(null);
-        }
-        setInfoWindowVisible(false);
-    };
+  const handlePlaceSelected = (place) => {
+    handleSetLocation({
+      lat: place.geometry.location.lat(),
+      lng: place.geometry.location.lng(),
+    });
+    setZoom(15);
+    setRecentSearches([
+      ...recentSearches,
+      {
+        name: place.name,
+        lat: place.geometry.location.lat(),
+        lng: place.geometry.location.lng(),
+      },
+    ]);
+  };
 
+  const handleSetLocation = (location) => {
+    setCurrentPosition({
+      lat: location.lat,
+      lng: location.lng,
+    });
+    setZoom(15);
+    setInfoWindowPosition({
+      lat: location.lat,
+      lng: location.lng,
+    });
+    setInfoWindowVisible(true);
+  };
 
-    const handleMapRightClick = (e) => {
-        handleSetLocation({
-            lat: e.latLng.lat(),
-            lng: e.latLng.lng(),
-        });
-    };
+  const handleMapClick = () => {
+    if (marker) {
+      marker.setMap(null);
+      setMarker(null);
+    }
+    setInfoWindowVisible(false);
+  };
 
-    const handleSaveLocation = () => {
-        const formattedLat = infoWindowPosition.lat.toFixed(2);
-        const formattedLng = infoWindowPosition.lng.toFixed(2);
-        const newLocation = {
-            name: `(${formattedLat}, ${formattedLng})`,
-            lat: infoWindowPosition.lat,
-            lng: infoWindowPosition.lng,
-        };
-        setSavedLocations([...savedLocations, newLocation]);
+  const handleMapRightClick = (e) => {
+    handleSetLocation({
```

Comment:
there is some glitch after we remove the infowindow. Will fix it after this PR is merged